ITR/data/base_providers.py

```diff
@@ -265,17 +296,20 @@ def _convert_projections_to_series(self, company: ICompanyData, feature: str,
                 )
             else:
                 projections = company_dict[feature][list(projection_scopes.keys())[0]]['projections']
-        return pd.Series(
-            {p['year']: p['value'] for p in projections},
-            name=company.company_id, dtype=f'pint[{emissions_units}/({production_units})]')
 
     def _calculate_target_projections(self, production_bm: BaseProviderProductionBenchmark):
         """
         We cannot calculate target projections until after we have loaded benchmark data.
         We do so when companies are associated with benchmarks, in the DataWarehouse construction
         
         :param production_bm: A Production Benchmark (multi-sector, single-scope, 2020-2050)
-        """
+        """        
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            # FIXME: Note that we don't need to call with a scope, because production is independent of scope.
+            df_pp = production_bm._get_projected_production()
+        df_partial_pp = df_pp.add(1.0).cumprod(axis=1)
```

Comment:
what does this line do?
an in-code comment could help

Reply:
If you are asking about line 311, this is the fundamental projection calculation.  The projected production is a year-over-year fractional increase (so 3% growth would be 0.03).  By adding 1 to each term and multiplying across the whole series, it generates a year-by-year projection of what the production will be.  Also, I note that the code snipped is out of date:

```
        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            # FIXME: Note that we don't need to call with a scope, because production is independent of scope.                                                                                                               
            # We use the arbitrary EScope.AnyScope just to be explicit about that.                                                                                                                                           
            df_pp = production_bm._get_projected_production(EScope.AnyScope)
	df_partial_pp = df_pp.add(1.0).cumprod(axis=1)
```
I'm ambivalent about adding a comment here.

Reply:
I'm talking about ’df_pp.add(1.0).cumprod(axis=1)‘.

It lack clarity, from my perspective.
I don't know how we will maintain it in future

Reply:
That one line is *the* calculation of projected production factor over time.  Multiply that by base production and you get cumulative emissions by year.  If you know how pdf.DataFrame.cumprod works, it should be self-evident.  If you don't, you should consult Pandas documentation.  But maybe you'd like to see a paragraph that spells out in a comment how this one line changes a dataframe of year-over-year production changes into a dataframe of total production quantities, perhaps by showing how the numbers change?

Reply:
i enter a function, which from the name i expect to calculate target projections.
that's fine.

step 1: get a dataframe of projected productions form benchmarks.
that's fine.

step 2: add 1.0 to every value in the dataframe, and get a cumulitive product over axis 1.
questions here:
* why adding 1?
* what are columns of an expected `df_pp`?
* what is axis 1?
* why is this action creates partial projected productions?
* why is to more useful, then just projected productions?

i ask you to add a comment, to make it possible for anybody else, do understand the behaviour by readyng the code, w/o running the code

Reply:
I just pushed changes with detailed comments.  Please let me know if that's what you were hoping to see.